examples/vscode-using-workspace-api/src/extension.ts

```diff
@@ -0,0 +1,28 @@
+import { ExtensionContext, extensions, window } from "vscode";
+import { BasToolkit } from "@sap-devx/app-studio-toolkit-types";
+
+async function activate(context: ExtensionContext): Promise<void> {
+  // Access the BAS Toolkit apis via vscode's `getExtension`
+  const basToolkitAPI: BasToolkit = extensions.getExtension(
+    "SAPOSS.app-studio-toolkit"
+  )?.exports;
+
+  const workspaceAPI = basToolkitAPI.workspaceAPI;
+  const outputChannel = window.createOutputChannel(context.extension.id);
+  const projects = await workspaceAPI.getProjects();
```

Comment:
The unresolved alert already existed in the code, I just moved the code around and I need to fix it
as part of a different PR that cleans up this commit https://github.com/SAP/app-studio-toolkit/commit/6027b6e65a0f953bd9d1d2de91e1e94c5086f3cb

Reply:
I am not sure about the suggested change:
`  const projects = await workspaceAPI?.getProjects() || [];`

1. `workspaceAPI` always exists on `basToolkitAPI`
2. if `getProjects()` fails horribly it would throw (reject) the promise so the `|| []` branch would not be entered.
3. if there are no projects then `getProjects()` will return an empty array.

Maybe you wanted to guard against `basToolkitAPI` not loading?
```typescript
  const basToolkitAPI: BasToolkit = extensions.getExtension(
    "SAPOSS.app-studio-toolkit"
  )?.exports;

  const workspaceAPI = basToolkitAPI.workspaceAPI;
```

I could add a guard condition that halts the extension's activation if the `extensions.getExtension(
    "SAPOSS.app-studio-toolkit"
  )?.exports` fails

Reply:
👍

Reply:
Although maybe this already automatically happens due to the `extensionDep` ?

- https://github.com/SAP/app-studio-toolkit/pull/100/files#diff-f50216efa297e3d3cba970120a57ee19085c1fe2d4a08bf2ab60dcfea2f15bb4R24-R26